### PR TITLE
[stable/prometheus-adapter] Added pod annotations to chart. 

### PR DESCRIPTION
**Pull request by aaronmell**
Upgraded chart to latest version of image

Signed-off-by: Aaron Mell <aaronamell@gmail.com>

#### What this PR does / why we need it:
I needed to add some pod annotations, which the chart doesn't support. Also, the version was outdated.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/prometheus-adapter/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: prometheus-adapter
-version: 1.2.0
+version: v1.2.1
```
Comment (steven-sheehy) [inline stable/prometheus-adapter/Chart.yaml]: `1.3.0` for new features. And drop the `v` for consistency with past releases.

File: stable/prometheus-adapter/templates/custom-metrics-apiserver-deployment.yaml [hunk 2]
```diff
@@ -23,6 +23,9 @@ spec:
       name: {{ template "k8s-prometheus-adapter.name" . }}
       annotations:
         checksum/config: {{ include (print $.Template.BasePath "/custom-metrics-configmap.yaml") . | sha256sum }}
+{{- with .Values.podAnnotations }}
+{{ toYaml . | indent 8 }}
```
Comment (steven-sheehy) [inline stable/prometheus-adapter/templates/custom-metrics-apiserver-deployment.yaml]: Use nindent like the rest of the file and align to be under `checksum/config`

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Comment by aaronmell** [comment]
@steven-sheehy Code review changes made

**Comment by steven-sheehy** [comment]
/lgtm
Thanks!

**Pull request merged**